src/main/java/cd/go/framework/ldap/Ldap.java

```diff
@@ -99,15 +101,23 @@ private static SearchControls getSimpleSearchControls(int maxResult) {
 
     private NamingEnumeration<SearchResult> search(String filter, Object[] filterArgs, int maxResult) throws NamingException {
         DirContext dirContext = getDirContext(ldapConfiguration, ldapConfiguration.getManagerDn(), ldapConfiguration.getPassword());
-        for (String base : ldapConfiguration.getSearchBases()) {
-            try {
+        try {
+            for (String base : ldapConfiguration.getSearchBases()) {
```

Comment:
Moving the try outside the loop would break few scenarios, e.g in case of multiple search base if a search with the first base errors out the rest of the search bases are ignored